server.go

```diff
@@ -1602,6 +1621,12 @@ func (s *Server) serveConn(c net.Conn) error {
 			break
 		}
 
+		if ctx.held {
```

Comment:
There is a data race here:

1) The ctx.held may be false here, so the ctx is re-used.
2) Then ctx.Hold() is called in a separate goroutine created by request handler.
3) Then the ctx is concurrently used by another request handler and the goroutine from the step 2. Data race detected :)

Why not using existing [RequestCtx.TimeoutError*](https://godoc.org/github.com/valyala/fasthttp#RequestCtx.TimeoutError) mechanism?

Personally, I don't like both approaches - the existing one and the approach from the PR, since they look fragile and racy.

I prefer passing only relevant copies of data to external goroutines from request handler. This automatically limits RequestCtx lifetime by request handler scope and avoids data races.